client/additional-methods-setup/methods-selector/add-payment-methods-task.js

```diff
@@ -58,18 +65,69 @@ const usePaymentMethodsCheckboxState = () => {
 };
 
 const AddPaymentMethodsTask = () => {
+	const availablePaymentMethods = useGetAvailablePaymentMethodIds();
+	const {
+		enabledPaymentMethodIds,
+		updateEnabledPaymentMethodIds,
+	} = useEnabledPaymentMethodIds();
+
+	const { saveSettings, isSaving } = useSettings();
+
+	// I am using internal state in this component
+	// and committing the changes on `enabledPaymentMethodIds` only when the "continue" button is clicked.
+	// Otherwise a user could navigate to another page via soft-routing and the settings would be in un-saved state,
+	// possibly causing errors.
```

Comment:
Wouldn't it better to store everything in the general state? If the user leaves the screen and comes back, they would find this component as they left it. I'm fine with leaving it as it is, but I usually try to avoid mixing internal state with application-wide state. What do you think?

Reply:
I agree with you that mixing internal state and redux state should be avoided when possible. However, in this case it would cause issues. I think it would be more serious if the soft routed to another page and the settings were inconsistent with what is actually stored server-side. They would think that they _already_ added some payment methods, but they really didn't.
And IMO the same would apply when going back to this page. They would add some payment methods, not save and go back to the unsaved state of the page. Only to realize later that the methods don't appear at checkout. LMK if I misunderstood :)

Reply:
Ok, you convinced me again! :smile: